### DISCUSSION TITLE
Azure Monitor: Query builder for Logs

### DISCUSSION THREAD

**Discussion by andresmgot** [grafana/grafana#56405] [Data Sources]
As a user, using Azure Monitor Logs has additional difficulty of having to learn Kusto Query Language (KQL) to monitor resources. To help with this, in [ADX](https://github.com/grafana/azure-data-explorer-datasource), there is a query builder for KQL that people can use to explore and create Kusto queries.

![Screenshot from 2022-10-05 16-55-53](https://user-images.githubusercontent.com/4025665/194094511-7bbe3a6d-daaf-4f06-897f-0922a75c6403.png)

My suggestion is to move the builder code to a shareable package between ADX and Azure Monitor so both can benefit from using a query builder.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
